azure.go

```diff
@@ -317,6 +366,20 @@ func (b *azureAuthBackend) getAzureSettings(ctx context.Context, config *azureCo
 		}
 	}
 
+	envName := os.Getenv("AZURE_ENVIRONMENT")
+	if envName == "" {
+		envName = config.Environment
+	}
+	if envName == "" {
+		settings.Environment = azure.PublicCloud
+	} else {
+		var err error
+		settings.Environment, err = azure.EnvironmentFromName(envName)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I wonder if this can be combined with the `os.Getenv("AZURE_ENVIRONMENT")` above on L355? I understand that `settings.Environment` and `settings.CloudConfig` take different values, but it appears a little redundant to read the same env var twice.

Reply:
Ah, good call. Simplified the logic a bit so we can combine the two operations